src/test/xmeta/test_symbol_table.cpp

```diff
@@ -110,3 +110,176 @@ TEST_CASE("Enum circular dependency")
+    REQUIRE(properties[0]->get_id() == "property_1");
+    REQUIRE(properties[1]->get_id() == "property_2");
+    REQUIRE(properties[2]->get_id() == "property_3");
+    REQUIRE(properties[3]->get_id() == "property_4");
+    REQUIRE(properties[4]->get_id() == "property_5");
+    REQUIRE(properties[0]->get_get_method() != nullptr);
+    REQUIRE(properties[0]->get_get_method()->get_id() == "get_property_1");
+    REQUIRE(properties[0]->get_set_method() != nullptr);
+    REQUIRE(properties[0]->get_set_method()->get_id() == "set_property_1");
+    REQUIRE(properties[1]->get_get_method() != nullptr);
+    REQUIRE(properties[1]->get_set_method() != nullptr);
+    REQUIRE(properties[1]->get_get_method()->get_id() == "get_property_2");
+    REQUIRE(properties[1]->get_set_method()->get_id() == "set_property_2");
+    REQUIRE(properties[1]->is_get_method_declared_first());
+    REQUIRE(properties[2]->get_get_method() != nullptr);
+    REQUIRE(properties[2]->get_set_method() != nullptr);
+    REQUIRE(properties[2]->get_get_method()->get_id() == "get_property_3");
+    REQUIRE(properties[2]->get_set_method()->get_id() == "set_property_3");
+    REQUIRE(!properties[2]->is_get_method_declared_first());
+    REQUIRE(properties[3]->get_get_method() != nullptr);
+    REQUIRE(properties[3]->get_set_method() != nullptr);
+    REQUIRE(properties[3]->get_get_method()->get_id() == "get_property_4");
+    REQUIRE(properties[3]->get_set_method()->get_id() == "set_property_4");
+    REQUIRE(properties[3]->is_get_method_declared_first());
+    REQUIRE(properties[4]->get_get_method() != nullptr);
+    REQUIRE(properties[4]->get_set_method() != nullptr);
+    REQUIRE(properties[4]->get_get_method()->get_id() == "get_property_5");
+    REQUIRE(properties[4]->get_set_method()->get_id() == "set_property_5");
+    REQUIRE(!properties[4]->is_get_method_declared_first());
+
+    auto classes = ns_body->get_classes();
+    REQUIRE(classes.size() == 1);
+    REQUIRE(classes.find("C") != classes.end());
+    auto c = classes.at("C");
+    properties = c->get_properties();
+    REQUIRE(properties.size() == 5);
+    REQUIRE(properties[0]->get_id() == "property_1");
+    REQUIRE(properties[1]->get_id() == "property_2");
+    REQUIRE(properties[2]->get_id() == "property_3");
+    REQUIRE(properties[3]->get_id() == "property_4");
+    REQUIRE(properties[4]->get_id() == "property_5");
+    REQUIRE(properties[0]->get_get_method() != nullptr);
+    REQUIRE(properties[0]->get_get_method()->get_id() == "get_property_1");
+    REQUIRE(properties[0]->get_set_method() != nullptr);
+    REQUIRE(properties[0]->get_set_method()->get_id() == "set_property_1");
+    REQUIRE(properties[1]->get_get_method() != nullptr);
+    REQUIRE(properties[1]->get_set_method() != nullptr);
+    REQUIRE(properties[1]->get_get_method()->get_id() == "get_property_2");
+    REQUIRE(properties[1]->get_set_method()->get_id() == "set_property_2");
+    REQUIRE(properties[1]->is_get_method_declared_first());
```

Comment:
I alluded to this elsewhere, but instead of this narrow check of whether a particular getter or setter is declared first, what we really care about is the ordering of the methods, particularly on the interface(s) this class implements. Check that ordering instead of this flag.